backend/src/modules/favourite/service/favourite/favourite.service.ts

```diff
@@ -0,0 +1,85 @@
+import { Injectable } from '@nestjs/common';
+import { InjectModel } from '@nestjs/mongoose';
+import { Model } from 'mongoose';
+
+import { Favourite, FavouriteItem } from '../../models/favourite.interface';
+import { Product } from 'src/modules/product/product.interface';
+
+
+@Injectable()
+export class FavouriteService {
+  constructor(
+    @InjectModel('Favourites') private readonly favouriteModel: Model<Favourite>,
+    @InjectModel('Product') private readonly productModel: Model<Product>
+  ) {}
+
+  public async getUserFavourites(userId: string) {
+    const userFavourites = await this.favouriteModel.findOne({userId});
+    if (userFavourites) {
+      const responseCart: {userId, items} = {
```

Comment:
dont think it is good idea to return userId. Caller know userId (he puts it to current function) and if it is needed for him he can construct such objects himself